src/snowflake/snowpark/mock/plan.py

```diff
@@ -496,7 +512,9 @@ def execute_mock_plan(
             for exp in source_plan.grouping_expressions
         ]
         for column_name, _, column_type in column_exps:
-            result_df_sf_Types[column_name] = column_type
+            result_df_sf_Types[
+                column_name
+            ] = column_type  # TODO: fix this, this does not work
```

Comment:
This  is blocking enabling `test_create_temp_view_on_functions`, essentially `col("a")` does not initialize the datatype of the column since it is an unresolved attribute. We probably need to `calculate_expression` to fix this.

Reply:
let's make sure we create a JIRA for this